Improvements

Aurelien30000 opened the pull request:
- Delete unused ``BlockDataSetter`` classes.
- Fix #47
- Completely replace Bukkit ``Material`` API by WE ``BlockTypes`` & ``ItemTypes``.
- Refactor ``BlockTracer`` using WE Api.
- Fix TreeSniper brush failing due to ``Async Block Remove``.

Aurelien30000 committed:
Improvements
- Delete unused ``BlockDataSetter`` classes.
- Fix https://github.com/IntellectualSites/FastAsyncVoxelSniper/issues/47
- Completely replace Bukkit ``Material`` API by WE ``BlockTypes`` & ``ItemTypes``.
- Refactor ``BlockTracer`` using WE Api.
- Fix TreeSniper brush failing due to ``Async Block Remove``.

NotMyFault commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/TreeSnipeBrush.java:
```diff
@@ -43,18 +43,21 @@ public void handleCommand(String[] parameters, Snipe snipe) {
         BlockState currentBlockData = getBlock(targetBlock.getX(), targetBlock.getY() - 1, targetBlock.getZ());
         setBlockType(targetBlock.getX(), targetBlock.getY() - 1, targetBlock.getZ(), BlockTypes.GRASS_BLOCK);
-        generateTree(this.treeType, targetBlock);
+        if (!generateTree(this.treeType, targetBlock)) {
```
(Wrong spot, but) maybe leave a note placing trees thru this brush is currently not undo-able, because spigot changed linked method?

Aurelien30000 replied:
Certainly! _Wrong spot too, u should also add a warning in FAWE then 🤔_

NotMyFault replied:
Good point, https://github.com/IntellectualSites/FastAsyncWorldEdit/commit/69067fe8c8a87ec1c5a1528f605724dc71d74e43

NotMyFault commented on src/main/java/com/thevoxelbox/voxelsniper/command/executor/VoxelReplaceExecutor.java:
```diff
@@ -54,22 +51,29 @@ public void executeCommand(CommandSender sender, String[] arguments) {
+            toolkitProperties.setReplaceBlockType(blockType);
+            messenger.sendReplaceBlockTypeMessage(blockType);
+        } else {
+            sender.sendMessage(ChatColor.RED + "You have entered an invalid Item ID.");
```
Item name > ID. FAVS doesn't accept IDs anylonger.

Aurelien30000 replied:
Makes more sense.

NotMyFault commented on src/main/java/com/thevoxelbox/voxelsniper/sniper/toolkit/BlockTracer.java:
```diff
@@ -1,43 +1,23 @@
-        Iterator<Block> iterator = new BlockIterator(player, distance);
-        iterate(iterator);
-    }
+    public BlockTracer(Player player, int distance, boolean useLastBlock) {//FAWE modify (make public)
```
We don't really need these annotations here anymore, or never really used them. Upstream VS is dead, can remove them on sight imo.

Aurelien30000 replied:
👍

Aurelien30000 committed:
Resolve requests
- Remove FAWE annotations, now senseless.
- Item IDs no longer exist, use Item names.
- Add a warning to not undo-able trees generated by tree-snipe brush.

Aurelien30000 committed:
Revert a moved line to a wrong place.

NotMyFault commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/TreeSnipeBrush.java:
```diff
@@ -72,6 +75,8 @@ public void sendInfo(Snipe snipe) {
         messenger.sendBrushNameMessage();
         printTreeType(messenger);
+        // TODO Remove once fixed
+        messenger.sendMessage(ChatColor.GOLD + "Warning: This brush is currently not undo-able due to a server bug!");
```
```suggestion
        messenger.sendMessage(ChatColor.RED + "Warning: This brush is currently not undo-able due to a Spigot bug!");
```

Aurelien30000 committed:
Update src/main/java/com/thevoxelbox/voxelsniper/brush/type/TreeSnipeBrush.java

Co-authored-by: NotMyFault <mc.cache@web.de>
